pkg/controller/kogitodataindex/resource/resource_comparator.go

```diff
@@ -32,15 +34,28 @@ func GetComparator() compare.MapComparator {
 			WithType(reflect.TypeOf(v1.Service{})).
 			WithCustomComparator(framework.CreateServiceComparator()).
 			UseDefaultComparator().
-			BuildAsFunc())
+			Build())
 
 	resourceComparator.SetComparator(
 		framework.
 			NewComparatorBuilder().
 			WithType(reflect.TypeOf(routev1.Route{})).
 			WithCustomComparator(framework.CreateRouteComparator()).
 			UseDefaultComparator().
-			BuildAsFunc())
+			Build())
+
+	resourceComparator.SetComparator(
+		framework.NewComparatorBuilder().
+			WithType(reflect.TypeOf(appsv1.Deployment{})).
+			UseDefaultComparator().
```

Comment:
It is not necessary to set a comparator if you want to use the default one.

Reply:
Yes, the funny thing is that I wrote the API and I forget about it. I might need to do some refactor to it.